code/game/supplyshuttle.dm

```diff
@@ -46,6 +46,192 @@ var/datum/controller/supply/supply_controller = new()
 
 //SUPPLY PACKS MOVED TO /code/defines/obj/supplypacks.dm
 
+/obj/docking_port/stationary/supply
+	id = "supply_home"
+	roundstart_template = /datum/map_template/shuttle/supply
+	width = 5
+	dwidth = 2
+	dheight = 2
+	height = 5
+/obj/docking_port/mobile/supply
+	name = "supply shuttle"
+	id = SHUTTLE_SUPPLY
+	callTime = 15 SECONDS
+
+	dir = WEST
+	port_direction = EAST
+	width = 5
+	dwidth = 2
+	dheight = 2
+	height = 5
+	movement_force = list("KNOCKDOWN" = 0, "THROW" = 0)
+	use_ripples = FALSE
+	var/list/gears = list()
+	var/list/obj/structure/machinery/door/poddoor/railing/railings = list()
+	///The faction of this docking port (aka, on which ship it is located)
+	var/faction = FACTION_MARINE
+	/// Id of the home docking port
+	var/home_id = "supply_home"
+
+/obj/docking_port/mobile/supply/Destroy(force)
+	for(var/i in railings)
+		var/obj/structure/machinery/door/poddoor/railing/railing = i
+		railing.linked_pad = null
+	railings.Cut()
+	return ..()
+
+
+/obj/docking_port/mobile/supply/afterShuttleMove()
+	. = ..()
+	if(getDockedId() == home_id)
+		for(var/j in railings)
+			var/obj/structure/machinery/door/poddoor/railing/R = j
+			R.open()
+
+/obj/docking_port/mobile/supply/on_ignition()
+	if(getDockedId() == home_id)
+		for(var/j in railings)
+			var/obj/structure/machinery/door/poddoor/railing/R = j
+			R.close()
+		for(var/i in gears)
+			var/obj/structure/machinery/gear/G = i
+			G.start_moving(NORTH)
+	else
+		for(var/i in gears)
+			var/obj/structure/machinery/gear/G = i
+			G.start_moving(SOUTH)
+
+/obj/docking_port/mobile/supply/register()
+	. = ..()
+	for(var/obj/structure/machinery/gear/G in GLOB.machines)
+		if(G.id == "supply_elevator_gear")
+			gears += G
+			RegisterSignal(G, COMSIG_PARENT_QDELETING, .proc/clean_gear)
+	for(var/obj/structure/machinery/door/poddoor/railing/R in GLOB.machines)
+		if(R.id == "supply_elevator_railing")
+			railings += R
+			RegisterSignal(R, COMSIG_PARENT_QDELETING, .proc/clean_railing)
+			R.linked_pad = src
+			R.open()
+
+///Signal handler when a gear is destroyed
+/obj/docking_port/mobile/supply/proc/clean_gear(datum/source)
+	SIGNAL_HANDLER
+	gears -= source
+
+///Signal handler when a railing is destroyed
+/obj/docking_port/mobile/supply/proc/clean_railing(datum/source)
+	SIGNAL_HANDLER
+	railings -= source
+
+/obj/docking_port/mobile/supply/canMove()
+	if(is_station_level(z))
+		return check_blacklist(shuttle_areas)
+	return ..()
+
+/obj/docking_port/mobile/supply/proc/check_blacklist(areaInstances)
```

Comment:
I would generally recommend trying to split this logic out. Conceptually I have tried to keep the shuttles code explicitly based around interacting with map stuff, other map objects as part of your transition. This logic could be pulled out into a datum just to improve cohesion